domain/network/service/interface.go

```diff
@@ -13,12 +13,19 @@ import (
 	"github.com/juju/juju/environs"
 )
 
-// Provider is the interface that the network service requires to be able to
+// ProviderWithNetworking is the interface that the network service requires to be able to
 // interact with the underlying provider.
-type Provider interface {
+type ProviderWithNetworking interface {
 	environs.Networking
 }
 
+// ProviderWithZones is the interface that the network service requires to be able to
+// interact with the underlying provider.
+type ProviderWithZones interface {
+	// AvailabilityZones returns all availability zones in the provider.
+	AvailabilityZones(ctx context.Context) (network.AvailabilityZones, error)
```

Comment:
Why can't we have only one provider interface like:
```
type Provider interface {
	environs.Networking
	// AvailabilityZones returns all availability zones in the provider.
	AvailabilityZones(ctx context.Context) (network.AvailabilityZones, error)
}
```
?
We have `AvailabilityZones()` on all the providers. I had to do something similar recently but it was because I needed some function only implemented on the k8s provider.